Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -615,7 +615,20 @@ public void PropertyChanged(object sender, PropertyChangedEventArgs args)
 					}
 				}
 
-				Device.BeginInvokeOnMainThread(() => _expression.Apply());
+				Action action = () => _expression.Apply();
+				if (_expression._weakTarget.TryGetTarget(out BindableObject obj) && obj is Element element && element.Dispatcher != null && element.Dispatcher.IsInvokeRequired)
```

Comment:
The `obj is Element` check is not required because `BindableObject` has a `Dispatcher`.